sw/device/tests/sha256_test.c

```diff
@@ -21,16 +27,54 @@ static const uint32_t kExpectedDigest[8] = {0xdc96c23d, 0xaf36e268, 0xcb68ff71,
 const test_config_t kTestConfig = {};
 
 bool test_main(void) {
-  LOG_INFO("Running SHA256 test");
+  dif_hmac_config_t config = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_HMAC_BASE_ADDR),
+      .message_endianness = kDifHmacEndiannessBig,
+      .digest_endianness = kDifHmacEndiannessBig,
+  };
+  CHECK(dif_hmac_init(&config, &hmac0) == kDifHmacOk);
+
+  CHECK(dif_hmac_mode_sha256_start(&hmac0) == kDifHmacOk);
+
+  const char *data = kData;
+  size_t data_left = kDataLen;
+  while (data_left > 0) {
+    size_t bytes_sent;
+    switch (dif_hmac_fifo_push(&hmac0, data, data_left, &bytes_sent)) {
+      case kDifHmacFifoOk:
+        goto push_end;
+      case kDifHmacFifoFull:
+        data += bytes_sent;
+        data_left -= bytes_sent;
+      default:
+        LOG_ERROR("Error while pushing to FIFO.");
+        return false;
+    }
+  }
+push_end:
 
-  uint32_t digest[8];
-  hw_SHA256_hash(kData, kDataLen, (uint8_t *)digest);
+  CHECK(dif_hmac_process(&hmac0) == kDifHmacOk);
+  dif_hmac_digest_t digest;
+  while (true) {
+    switch (dif_hmac_digest_read(&hmac0, &digest)) {
+      case kDifHmacDigestProcessing:
+        continue;
+      case kDifHmacDigestOk:
+        goto read_end;
+      default:
+        LOG_ERROR("Error while reading the digest.");
+        return false;
+    }
+  }
+read_end:
 
-  for (uint32_t i = 0; i < 8; i++) {
-    if (digest[i] != kExpectedDigest[i]) {
-      LOG_ERROR("Digest mismatched at index %d: exp: %x, act: %x", i, digest[i],
-                kExpectedDigest[i]);
-      flash_write_scratch_reg(digest[i]);
+  for (uint32_t i = 0; i < 8; ++i) {
+    uint32_t got = digest.digest[i];
+    uint32_t want = kExpectedDigest[i];
+    if (got != want) {
+      LOG_ERROR("Digest mismatch at index %d: wanted 0x%x, got 0x%x.", i, want,
+                got);
+      flash_write_scratch_reg(got);  // ?
```

Comment:
The question mark here isn't a useful comment. What about adding a TODO for documenting this?

Reply:
Done. I have *no idea* what this is here for, but I'm not going to delete this quite yet.